field.go

```diff
@@ -96,8 +95,8 @@ func String(key string, val string) Field {
 	return Field{key: key, fieldType: stringType, str: val}
 }
 
-// Stringer constructs a Field with the given key and value. The value
-// is the result of the String method.
+// Stringer constructs a Field with the given key and the output of the value's
+// String method.
```

Comment:
Document that this is lazy and the `String` method is only called if/when the entry is written.